### ISSUE TITLE
Client deleted a full sync directory which is still on the server

### ISSUE THREAD

**Issue by cedricpradalier** [owncloud/client#3124]
Hi,

I'm running my own owncloud server (linux 12.04/postgresql/php 5.3.10), and synchronizing a laptop running Ubuntu 14.04 with it, using the desktop client (1.8). The server is behind a firewall and the laptop can only connect to it while on the same network. I'm running OC 7 because OC 8 would require a switch of PHP version. 

I recently had a weird behaviour I can neither reproduce nor explain: the client decided unilaterally do delete most of the files (~9GB) in one of the synchronized folders. The files were and are still on the server and visible with the WebDav interface. The deleted files could not be found in the Trash folder.

For all athe deleted files, the client log contains the following, so the client is doing it on purpose, except the files are still on the server.
|0|path/to/file.                                                 pdf|INST_REMOVE|Down|1388746608|5519573725631|2342287||4||0|0|0|||INST_NONE|
|0|path/to/file.                                                 pptx|INST_REMOVE|Down|1388746606|5519573762a97|4358891||4||0|0|0|||INST_NONE|

The owncloud client reported that the synchronization of this folder was sucessful, although the folder remained empty (expect the ignored files). Later lines in the client log show the following even though the folder is empty.
#=#=#=# Syncrun started 2015-04-15T14:21:23 until 2015-04-15T14:21:26 (2906     msec)
#=#=#=# Syncrun started 2015-04-15T16:22:04 until 2015-04-15T16:22:06 (2041     msec)
#=#=#=# Syncrun started 2015-04-15T18:22:40 until 2015-04-15T18:22:42 (2216     msec)
There are no errors in the log, no error in the activity report, and the files are not reported as deleted on the server activity report. The folder is NOT synchronized by another sync service. 

I ended up using the owncloudcmd interface to redownload a copy of the folder locally, got all my files, disconnected owncloud, replaced the weird folder by the newly downloaded one and reconnected owncloud. The new folder is now synchronizing correctly. I saved the old folder for now in case there is some data in it that could be useful for debugging.

Because I can't reproduce the case, I'm not expecting a fix, just reporting in case somebody else is faced with the same issue.

**Comment by RealRancor** [comment]
Hi,

after reading it here again i'm not sure but it seems similar to:

https://github.com/owncloud/client/issues/3116 and https://github.com/owncloud/client/issues/3102

**Comment by guruz** [comment]
@cedricpradalier Thanks, yes, this is still unclear currently. Could you also check the owncloud.log and server access_log for the things mentioned in the other reports?

**Comment by cedricpradalier** [comment]
I checked the mentioned possible duplicate but it does not seem to be exactly the same. 

First the missing files are not random and for the second report, there is a single client syncing with this server and the files are just not downloaded.

When checking the apache logs I don't see any PROFIND returning something different than 207, and the missing files have not been "DELETED". 

In the owncloud.log file on the server, there are many errors such as 
{"app":"hook","message":"error while running hook (OCA\\Search_Lucene\\Hooks::indexFile): An exception occurred while executing 'INSERT INTO \"oc_jobs\"(\"class\", \"argument\", \"last_run\") VALUES(?, ?, 0)':                 \n\nSQLSTATE[22001]: String data, right truncated: 7 ERROR:  value too long for type character varying(256)","level":3,"time":"2015-04-02T12:11:26+00:00"}

And from times to times (Every 500 lines or so):

{"app":"remote","message":"An exception occurred while executing 'INSERT INTO \"oc_filecache\" (\"mimepart\", \"mimetype\", \"mtime\", \"size\", \"etag\", \"storage_mtime\", \"permissions\", \"parent\", \"path_hash\",         \"path\", \"name\", \"storage\") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)':\n\nSQLSTATE[23505]: Unique violation: 7 ERROR:  duplicate key value violates unique constraint \"fs_storage_path_hash\"\nDETAIL:  Key (storage,    path_hash)=(1, 060b74346dcef7e1ad3e0d7fa29e7760) already exists.","level":4,"time":"2015-04-01T15:15:42+00:00"}
{"app":"PHP","message":"Session has been closed - no further changes to the session as allowed at \/var\/www\/owncloud\/lib\/private\/session\/memory.php#80","level":3,"time":"2015-04-01T15:15:42+00:00"}
{"app":"remote","message":"An exception occurred while executing 'INSERT INTO \"oc_filecache\" (\"mimepart\", \"mimetype\", \"mtime\", \"size\", \"etag\", \"storage_mtime\", \"permissions\", \"parent\", \"path_hash\",         \"path\", \"name\", \"storage\") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)':\n\nSQLSTATE[23505]: Unique violation: 7 ERROR:  duplicate key value violates unique constraint \"fs_storage_path_hash\"\nDETAIL:  Key (storage,    path_hash)=(1, bb4ea8bc29e65d68833f5c0d00eab92c) already exists.","level":4,"time":"2015-04-01T15:54:08+00:00"}
{"app":"PHP","message":"Session has been closed - no further changes to the session as allowed at \/var\/www\/owncloud\/lib\/private\/session\/memory.php#80","level":3,"time":"2015-04-01T15:54:08+00:00"}

**Comment by PVince81** [comment]
Is your data directory exclusive to ownCloud or are other apps/people accessing it directly outside of ownCloud ? Is your data directory mounted on NFS or some other backend or is it a local partition ?
Just asking this to exclude potential concurrent scanner issues in case the mtimes would change directly on disk.

**Comment by cedricpradalier** [comment]
The data directory is on a local partition on the server (this is a software RAID1 setup) and I am the only user of this owncloud server, no other app should be accessing the owncloud directories.

**Comment by DeepDiver1975** [comment]
> {"app":"remote","message":"An exception occurred while executing 'INSERT INTO \"oc_filecache\" (\"mimepart\", \"mimetype\", \"mtime\", \"size\", \"etag\", \"storage_mtime\", \"permissions\", \"parent\", \"path_hash\", \"path\", \"name\", \"storage\") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)':\n\nSQLSTATE[23505]: Unique violation: 7 ERROR: duplicate key value violates unique constraint \"fs_storage_path_hash\"\nDETAIL: Key (storage, path_hash)=(1, 060b74346dcef7e1ad3e0d7fa29e7760) already exists.","level":4,"time":"2015-04-01T15:15:42+00:00"}

these constrain issues will be fixed with 8.0.3 and 8.1

**Comment by cedricpradalier** [comment]
Is the constraint issue resulting in the client not syncing the directory? Or is it a separate issue?

**Comment by guruz** [comment]
@cedricpradalier Unfortunately so far. client 1.8.1 also contains fixes to make the discovery more robust, so that could fix it too.

**Comment by cedricpradalier** [comment]
Sorry if I'm a bit dense here, but does that mean I should close the issue?

**Comment by guruz** [comment]
If this problem ever happens again..

Can you please try to use something like:
```curl -X PROPFIND https://user:pwd@cloud.example.com/remote.php/webdav/your/path/where/there/is/files/which/dont/show/up/on/client |tidy -xml -i``` (tidy part us optional)
to see if the server returns the files which are missing on your desktop?

If you afterwards close your client, delete ```.csync_journal.db*``` and then start client again, will the files come back?

Can you also try to use ```beta1``` from now on? We have some fixes in there which could help. https://owncloud.org/install/#testing-development

**Comment by 1stone** [comment]
Maybe a helpfull side note:
I just encountered the same problem as described by cedricpradalier.
In my case I reckoned that there was a network failure with high package loss.
Just out of nowhere the OC client (Version 1.8.0 build 2139) erased suddenly the entire local sync dir.
When looking on the server, all files were still present there.
After the network has become stable again, the client downloaded all files again without any action on my side.

May this be an issue of some weird failure condition that the client does not handle properly?

**Comment by guruz** [comment]
@1stone Thanks, that's a good hint. It should be fixed with 1.8.1

**Comment by cedricpradalier** [comment]
This is likely to be the cause here as well: I dont necessarily stop all network traffic before undocking my laptop, so this could be seen as a major network failure.

**Comment by PVince81** [comment]
Do you guys use php-fpm ? AFAIK when the connection drops, php-fpm kills the PHP process. So if it's in the middle of an operation, it could break...

**Comment by moscicki** [comment]
@PVince, @dragotin : shouldn't  the sync client decide to delete file only after receiving a complete and well-formatted PROPFIND response with a 2xx code? In all other cases it should not take any decisions such as delete. So any network cut in the middle of the operation should not have impact. If this happened then it is either a client bug (some race condition for example) or a wierd server problem which would give a false but complete and well-formatted response. How likely is that?

Or am I missing anything?

@guruz: which commit fixes this in 1.8.1?

**Comment by dragotin** [comment]
@moscicki there was an error condition not properly caught in 1.8.0 but that is already fixed in 1.8.1

**Comment by ckamm** [comment]
@moscicki I think the patches were 0634a4d0c6eec0ff14f1c9df4d72f73ad809250d 2dbd27af76f748ebfdcbcd7954005075c878b2da 9d5f5ea3bc10772f6a803aa4c1e66b5a0105e242 and 9ffacd4ecdb60f5146cc628bf53acf74b9264dd4.

**Comment by moscicki** [comment]
@ckamm: thanks. Ouch! I think I see the point...

**Comment by guruz** [comment]
https://owncloud.org/install/#testing
The rc2 has code to automatically bring back the files. Sorry for the trouble.
